### PR TITLE
Explicitly check if NASM installed/reachable

### PR DESCRIPTION
**Pull request by master-of-zen**
Currently, it will just say that NASM version too old if NASM doesn't exist in the system.
This PR checks if NASM exists before checking its version.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -286,6 +286,18 @@ fn test_parse_nasm_version() {
...
 }
 
+#[cfg(feature = "asm")]
+fn is_nasm_installed_check() {
+  let mut cmd = std::process::Command::new("nasm");
+  cmd.arg("-h");
+  match cmd.output() {
+    Ok(_) => (),
+    Err(_) => {
+      panic!("Couldn't find NASM, make sure NASM installed and in PATH")
```
Comment (lu-zero) [inline build.rs:296]: Might be good to mention that is possible to not set the `asm` feature to avoid installing it.

Comment (master-of-zen) [reply build.rs]: Corrected

File: build.rs [hunk 2]
```diff
@@ -286,6 +286,20 @@ fn test_parse_nasm_version() {
   assert_eq!((2, 14, 0), parse_nasm_version(ver_str));
 }
 
+#[cfg(feature = "asm")]
+fn is_nasm_installed_check() {
+  let mut cmd = std::process::Command::new("nasm");
+  cmd.arg("-h");
+  match cmd.output() {
```
Comment (shssoichiro) [inline build.rs:293]: This is certainly minor in the overall scheme of things, but you can use `cmd.status()` instead which will be more performant as it avoids capturing stdout and stderr.

Comment (master-of-zen) [reply build.rs]: Corrected

**Review by shssoichiro — APPROVED** [review]

**Review by tdaede — APPROVED** [review]
It might be nice to emphasize that installing nasm will yield a huge speedup over disabling asm in the message.

**Pull request merged**